framework/src/node/state_machine/types.ts

```diff
@@ -15,7 +15,10 @@ import { Transaction } from '@liskhq/lisk-chain';
 import { IterateOptions } from '@liskhq/lisk-chain/dist-node/state_store';
 import { Schema } from '@liskhq/lisk-codec';
 import { Logger } from '../../logger';
-import { EventQueue } from './event_queue';
+
+export interface EventQueueAdder {
```

Comment:
Maybe generic name? `EventQueueManager` or `EventQueueUtil`

Reply:
I'm not sure as an interface those name would be suited because it sounds like it has actual implementation.
The main goal here is to limit the scope to `add` function for the modules. (and adding extra logic in state machine)